controllers/helmchart_controller.go

```diff
@@ -86,6 +86,10 @@ func (r *HelmChartReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{Requeue: true}, err
 	}
 
+	// set ownership reference so chart is garbage collected on
+	// repository removal
+	r.setOwnerRef(ctx, &chart, repository)
```

Comment:
**Note**: this requires an RBAC rule for `helmcharts/finalizers` on OpenShift. We should create a Kustomize patch for RBAC so that kubebuilder will not override it.